newIDE/app/src/GameDashboard/LeaderboardAdmin/LeaderboardAppearanceDialog.js

```diff
@@ -0,0 +1,330 @@
+      : ''
+  );
+  const [suffix, setSuffix] = React.useState<string>(
+    leaderboardCustomizationSettings &&
+      leaderboardCustomizationSettings.scoreFormatting.type === 'custom'
+      ? leaderboardCustomizationSettings.scoreFormatting.suffix
+      : ''
+  );
+  const [precision, setPrecision] = React.useState<number>(
+    leaderboardCustomizationSettings &&
+      leaderboardCustomizationSettings.scoreFormatting.type === 'custom'
+      ? leaderboardCustomizationSettings.scoreFormatting.precision
+      : 0
+  );
+  const [units, setUnits] = React.useState<string>(
+    leaderboardCustomizationSettings &&
+      leaderboardCustomizationSettings.scoreFormatting.type === 'time'
+      ? getIdentifierFromUnits(
+          leaderboardCustomizationSettings.scoreFormatting.units
+        )
+      : getIdentifierFromUnits(['second', 'millisecond'])
+  );
+  const [scorePreview, setScorePreview] = React.useState<number>(15.2);
+
+  const onSaveSettings = async (i18n: I18nType) => {
+    if (!scoreTitle) {
+      setScoreTitleError(i18n._(t`Title cannot be empty.`));
+      return;
+    }
+    setIsLoading(true);
+    const customizationSettings = {
+      scoreTitle,
+      scoreFormatting:
+        scoreType === 'custom'
+          ? {
+              type: scoreType,
+              prefix,
+              suffix,
+              precision,
+            }
+          : { type: scoreType, units: unitSelectOptions[units] },
+    };
+    await onSave(customizationSettings);
+  };
+
+  return (
+    <I18n>
+      {({ i18n }) => (
+        <Dialog
+          open={open}
+          maxWidth="sm"
+          onRequestClose={() => {
+            if (!isLoading) onClose();
+          }}
+          actions={[
+            <FlatButton
+              label={<Trans>Cancel</Trans>}
+              disabled={isLoading}
+              onClick={onClose}
+              key={'cancel'}
+            />,
+            <RaisedButton
+              primary
+              label={<Trans>Save</Trans>}
+              disabled={isLoading}
+              onClick={() => onSaveSettings(i18n)}
+              key={'save'}
+            />,
+          ]}
+        >
+          <Line>
+            <TextField
+              fullWidth
+              floatingLabelText={<Trans>Score column title</Trans>}
+              maxLength={20}
+              errorText={scoreTitleError}
+              value={scoreTitle}
+              onChange={(e, newTitle) => {
+                if (!!scoreTitleError && !!newTitle) setScoreTitleError(null);
+                setScoreTitle(newTitle);
+              }}
+            />
+          </Line>
+          <Text size="title">
+            <Trans>Score formatting</Trans>
+          </Text>
+          <Column>
+            <Line>
+              <SelectField
+                fullWidth
+                value={scoreType}
+                floatingLabelText={<Trans>Score display</Trans>}
+                onChange={(e, i, newValue) =>
+                  // $FlowIgnore
```

Comment:
should we instead do a `if (newValue !== 'custom' && newValue !== 'time') setScoreType('time')}` ?

I'm not sure this is that useful tbh, as it's checked in the backend too.

Reply:
Yes, I think that it would make the code heavy for something that is safe already